app/src/main/java/com/github/se/eventradar/model/repository/event/IEventRepository.kt

```diff
@@ -15,4 +16,8 @@ interface IEventRepository {
   suspend fun deleteEvent(event: Event): Resource<Unit>
 
   suspend fun getEventsByIds(ids: List<String>): Resource<List<Event>>
+
+  fun observeEvents(): Flow<Resource<List<Event>>>
+
+  fun observeUpcomingEvents(userId: String): Flow<Resource<List<Event>>>
```

Comment:
Yes this is great! The function signatures are exactly what I had in mind when discussing this solution last Friday! =)